robot/rospackages/src/mcu_control/scripts/MockPdsNode.py

```diff
@@ -0,0 +1,199 @@
+
+    #If parameters is an array, return the value associated for each parameter state
+    else:
+        outputArray = []
+        for x in range(len(parameters)):
+            paramState = get_param_exist(parameters[x])
+            if (paramState == "rise"):
+                outputArray.append(values.get("rise"))
+            elif (paramState == "stable"):
+                outputArray.append(values.get("stable"))
+            elif (paramState == "fall"):
+                outputArray.append(values.get("fall"))
+            else:
+                rospy.logwarn("Failed: Invalid parameter entered for {}".format(parameters[x]))
+        return outputArray
+
+def get_param_exist(parameter):
+    """Get parameter state if parameter exists and sets to None if it does not exist"""
+
+    try:
+        parameterState = rospy.get_param(parameter)
+    except:
+        rospy.logwarn("Parameter state None assigned to {}.".format(parameter))
+        parameterState = None
+    return parameterState
+
+def publish_mock_data(voltages, temps, currents, noiseValues, riseFallRates):
+    """Get and set parameters to be published"""
+
+    #Initialize node
+    node_name = "mock_pds_node"
+    rospy.init_node(node_name, anonymous=True)
+    rospy.loginfo("Initialized: {} node for mock PDS values".format(node_name))
+    rospyRate = 10
+    rate = rospy.Rate(rospyRate)
+
+    #Initialize publishers (Voltage, ThermistorTemps, Currents)
+    voltage = Voltage()
+    thermistorTemps = ThermistorTemps()
+    wheelCurrents = Currents()
+    pubVoltage = rospy.Publisher('battery_voltage', Voltage, queue_size=10)
+    pubTemp = rospy.Publisher('battery_temps', ThermistorTemps, queue_size=10)
+    pubWheelCurrent = rospy.Publisher('wheel_motor_currents', Currents, queue_size=10)
+
+    #Initialize 1 starting rise fall rate
+    currentRate = get_states_values("PDS_mock_rise_fall_rate", riseFallRates)
+    rospy.logwarn("currentRate: {}".format(currentRate))
+
+    #Initialize 1 starting noise
+    currentNoise = get_states_values("PDS_mock_global_noise", noiseValues)
+    rospy.logwarn("currentNoise: {}".format(currentNoise))
+
+    #Initialize 1 starting voltage
+    currentVoltage = get_states_values("PDS_mock_global_noise", voltages)
+    rospy.logwarn("currentVoltage: {}".format(currentVoltage))
+
+    #Initialize 3 starting temperatures
+    currentTemps = []
+    parameterTemps = ["PDS_mock_temp1", "PDS_mock_temp2", "PDS_mock_temp3"]
+    currentTemps = get_states_values(parameterTemps, temps)
+    rospy.logwarn("currentTemps: {}".format(currentTemps))
+
+    #Initialize 6 starting wheel currents
+    currentWheelCurrents = []
+    parameterCurrents = ["PDS_mock_wheel1_current", "PDS_mock_wheel2_current", "PDS_mock_wheel3_current", "PDS_mock_wheel4_current", "PDS_mock_wheel5_current", "PDS_mock_wheel6_current"]
+    currentWheelCurrents = get_states_values(parameterCurrents, currents)
+    rospy.logwarn("currentWheelCurrents: {}".format(currentWheelCurrents))
```

Comment:
```suggestion
    rospy.loginfo("currentRate: {}".format(currentRate))

    #Initialize 1 starting noise
    currentNoise = get_states_values("PDS_mock_global_noise", noiseValues)
    rospy.loginfo("currentNoise: {}".format(currentNoise))

    #Initialize 1 starting voltage
    currentVoltage = get_states_values("PDS_mock_global_noise", voltages)
    rospy.loginfo("currentVoltage: {}".format(currentVoltage))

    #Initialize 3 starting temperatures
    currentTemps = []
    parameterTemps = ["PDS_mock_temp1", "PDS_mock_temp2", "PDS_mock_temp3"]
    currentTemps = get_states_values(parameterTemps, temps)
    rospy.loginfo("currentTemps: {}".format(currentTemps))

    #Initialize 6 starting wheel currents
    currentWheelCurrents = []
    parameterCurrents = ["PDS_mock_wheel1_current", "PDS_mock_wheel2_current", "PDS_mock_wheel3_current", "PDS_mock_wheel4_current", "PDS_mock_wheel5_current", "PDS_mock_wheel6_current"]
    currentWheelCurrents = get_states_values(parameterCurrents, currents)
    rospy.loginfo("currentWheelCurrents: {}".format(currentWheelCurrents))
```

![image](https://user-images.githubusercontent.com/18620025/94352194-1410af00-0030-11eb-99fa-867db86ca98d.png)

Me after seeing all these warnings 
![image](https://user-images.githubusercontent.com/18620025/94352197-1d9a1700-0030-11eb-969b-583ac4427d12.png)

Reply:
tagging @AFknUsername because he will :eye: the meme and lol from his phone. then maybe review the PR

Reply:
btw @ginywiny when you accept this don't forget to add the "[#284]" because it doesnt use the hook